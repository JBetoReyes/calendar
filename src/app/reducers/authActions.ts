export const AUTH_CHECKING = '[AUTH] Checking login state.';
export const AUTH_CHECKED = '[AUTH] Checked.';
export const START_LOGIN = '[AUTH] Start login.';
export const LOGIN = '[AUTH] login.';
export const START_REGISTER = '[AUTH] Start register.';
export const TOKEN_RENEW = '[AUTH] Token renew.';
export const LOGOUT = '[AUTH] Logout.';

interface IAuthChecking {
  type: typeof AUTH_CHECKING;
}

interface IAuthCheckingFinish {
  type: typeof AUTH_CHECKED;
}

export type AuthActionsType = IAuthChecking | IAuthCheckingFinish;

export const startCheckAuth = (): AuthActionsType => {
  return {
    type: AUTH_CHECKING,
  };
};

export const authChecked = (): AuthActionsType => {
  return {
    type: AUTH_CHECKED,
  };
};
