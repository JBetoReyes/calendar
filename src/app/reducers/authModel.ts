export interface IAuthState {
  checking: boolean;
  uid: string | null;
  name: string | null;
}
