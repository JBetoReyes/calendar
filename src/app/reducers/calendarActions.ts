import { IAppCalendarEvent } from '../components/calendar/CalendarModel';

export const SET_ACTIVE_EVENT = '[CALENDAR] Set Active Event';
export const ADD_ACTIVE_EVENT = '[CALENDAR] Add Active Event';
export const UPDATE_EVENT = '[CALENDAR] Update Event';

interface ISetActiveEvent {
  type: typeof SET_ACTIVE_EVENT;
  payload: IAppCalendarEvent | null;
}

interface IAddEvent {
  type: typeof ADD_ACTIVE_EVENT;
  payload: IAppCalendarEvent;
}

interface IEventUpdated {
  type: typeof UPDATE_EVENT;
  payload: IAppCalendarEvent;
}

export type CalendarActionsTypes = ISetActiveEvent | IAddEvent | IEventUpdated;

export const setActiveEvent = (
  event: IAppCalendarEvent | null,
): CalendarActionsTypes => {
  return {
    type: SET_ACTIVE_EVENT,
    payload: event,
  };
};

export const addActiveEvent = (
  event: IAppCalendarEvent,
): CalendarActionsTypes => {
  return {
    type: ADD_ACTIVE_EVENT,
    payload: event,
  };
};

export const updateEvent = (event: IAppCalendarEvent): CalendarActionsTypes => {
  return {
    type: UPDATE_EVENT,
    payload: event,
  };
};
