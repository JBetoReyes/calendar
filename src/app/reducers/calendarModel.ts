import { IAppCalendarEvent } from '../components/calendar/CalendarModel';

export interface ICalendarState {
  events: IAppCalendarEvent[];
  activeEvent: IAppCalendarEvent | null;
}
