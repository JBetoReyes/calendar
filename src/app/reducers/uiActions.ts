export const OPEN_MODAL = '[UI] Open Modal';
export const CLOSE_MODAL = '[UI] Close Modal';

interface IOpenModal {
  type: typeof OPEN_MODAL;
}

interface ICloseModal {
  type: typeof CLOSE_MODAL;
}

export type UIActionsTypes = IOpenModal | ICloseModal;

export const openModal = (): UIActionsTypes => {
  return {
    type: OPEN_MODAL,
  };
};

export const closeModal = (): UIActionsTypes => {
  return {
    type: CLOSE_MODAL,
  };
};
