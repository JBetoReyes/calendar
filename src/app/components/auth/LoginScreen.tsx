/* eslint-disable prettier/prettier */
import React from 'react';
import Swal from 'sweetalert2';
import {connect} from 'react-redux';
import {AppSubmitEvent} from '../../../typings/htmlEvents';
import {startLogin, startRegister} from '../../reducers/authActions';
import useForm from '../../hooks/useForm';
import {IStoreState} from '../../store/storeModel';
import './LoginScreen.scss';

type LoginUserType = {
  lEmail: string;
  lPassword: string;
};

type RegisterUserType = {
  rName: string;
  rEmail: string;
  rPassword: string;
  rConfirmPassword: string;
};

const mapDispatchToProps = {
  startLogin,
  startRegister,
};

type MapDispatchToPropsType = typeof mapDispatchToProps;

type MyProps = Record<string, any>;
type Props = MyProps & MapDispatchToPropsType;

export const LoginScreen = (props: MyProps): JSX.Element => {
  const {formData: formDataLogin, handleChange: handleLoginChange} = useForm<
  LoginUserType
  >({
    lEmail: '',
    lPassword: '',
  });
  const {lEmail, lPassword} = formDataLogin;
  const {
    formData: formDataRegister,
    handleChange: handleRegisterChange,
  } = useForm<RegisterUserType>({
    rName: '',
    rEmail: '',
    rPassword: '',
    rConfirmPassword: '',
  });
  const {rName, rEmail, rPassword, rConfirmPassword} = formDataRegister;
  const {
    startLogin: dispatchStartLogin,
    startRegister: dispatchStartRegister,
  } = props as Props;
  const handleLogin = (e: AppSubmitEvent) => {
    e.preventDefault();
    dispatchStartLogin(lEmail, lPassword);
  };
  const handelRegister = (e: AppSubmitEvent) => {
    e.preventDefault();
    if (rPassword === rConfirmPassword) {
      dispatchStartRegister(rName, rEmail, rPassword);
    } else {
      Swal.fire('Error', 'Password should match.', 'error');
    }
  };
  return (
    <div className="container">
      <div className="row">
        <div className="login-container">
          <div className="col-md-6 login-form-1 h-75">
            <h3>Login</h3>
            <form onSubmit={handleLogin}>
              <div className="form-group">
                <input
                  type="text"
                  className="form-control"
                  placeholder="Email"
                  name="lEmail"
                  value={lEmail}
                  onChange={handleLoginChange}
                />
              </div>
              <div className="form-group">
                <input
                  type="password"
                  className="form-control"
                  placeholder="Password"
                  name="lPassword"
                  value={lPassword}
                  onChange={handleLoginChange}
                />
              </div>
              <div className="form-group">
                <input type="submit" className="btnSubmit" value="Login" />
              </div>
            </form>
          </div>

          <div className="col-md-6 login-form-2 h-75">
            <h3>Register</h3>
            <form onSubmit={handelRegister}>
              <div className="form-group">
                <input
                  type="text"
                  className="form-control"
                  placeholder="Name"
                  name="rName"
                  value={rName}
                  onChange={handleRegisterChange}
                />
              </div>
              <div className="form-group">
                <input
                  type="email"
                  className="form-control"
                  placeholder="Email"
                  name="rEmail"
                  value={rEmail}
                  onChange={handleRegisterChange}
                />
              </div>
              <div className="form-group">
                <input
                  type="password"
                  className="form-control"
                  placeholder="Password"
                  name="rPassword"
                  value={rPassword}
                  onChange={handleRegisterChange}
                />
              </div>

              <div className="form-group">
                <input
                  type="password"
                  className="form-control"
                  placeholder="Confirm your password"
                  name="rConfirmPassword"
                  value={rConfirmPassword}
                  onChange={handleRegisterChange}
                />
              </div>

              <div className="form-group">
                <input type="submit" className="btnSubmit" value="Register" />
              </div>
            </form>
          </div>
        </div>
      </div>
    </div>
  );
};

LoginScreen.propTypes = {};

export default connect<null, MapDispatchToPropsType, MyProps, IStoreState>(
  null,
  mapDispatchToProps,
)(LoginScreen);
