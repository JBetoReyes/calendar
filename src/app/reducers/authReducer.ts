import {AuthActionsType} from './authActions';
import {IAuthState} from './authModel';

const initialValue: IAuthState = {
  checking: true,
  uid: null,
  name: null,
  email: null,
};

const AuthReducer = (
  state: IAuthState = initialValue,
  action: AuthActionsType,
): IAuthState => {
  switch (action.type) {
    case '[AUTH] Login.':
      return {
        ...state,
        ...action.payload,
      };
    default:
      return state;
  }
};

export default AuthReducer;
