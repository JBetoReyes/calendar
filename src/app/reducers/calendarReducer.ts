import moment from 'moment';
import {
  ADD_ACTIVE_EVENT,
  CalendarActionsTypes,
  DELETE_EVENT,
  SET_ACTIVE_EVENT,
  UPDATE_EVENT,
} from './calendarActions';
import {ICalendarState} from './calendarModel';

const initialValue: ICalendarState = {
  events: [],
  activeEvent: null,
};

const CalendarReducer = (
  state: ICalendarState = initialValue,
  action: CalendarActionsTypes,
): ICalendarState => {
  switch (action.type) {
    case SET_ACTIVE_EVENT:
      return {
        ...state,
        activeEvent: action.payload,
      };
    case ADD_ACTIVE_EVENT:
      return {
        ...state,
        events: [...state.events, action.payload],
      };
    case UPDATE_EVENT:
      return {
        ...state,
        events: state.events.map((event) => {
          if (event.id === action.payload.id) {
            return {
              ...action.payload,
            };
          }
          return {
            ...event,
          };
        }),
      };
    case DELETE_EVENT:
      return {
        activeEvent: null,
        events: state.events.filter((event) => event.id !== action.payload),
      };
    default:
      return state;
  }
};

export default CalendarReducer;
