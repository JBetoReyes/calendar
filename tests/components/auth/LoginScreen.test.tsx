import configureStore from 'redux-mock-store';
import thunk from 'redux-thunk';
import {mount, ReactWrapper} from 'enzyme';
import React from 'react';
import {Provider} from 'react-redux';
import Swal from 'sweetalert2';

// eslint-disable-next-line import/no-named-as-default
import LoginScreen from '../../../src/app/components/auth/LoginScreen';
import {startLogin, startRegister} from '../../../src/app/reducers/authActions';

jest.mock('../../../src/app/reducers/authActions', () => {
  return {
    startLogin: jest.fn(),
    startRegister: jest.fn(),
  };
});

jest.mock('sweetalert2', () => ({
  fire: jest.fn(),
}));

const middleware = [thunk];
const mockStoreProvider = configureStore(middleware);
const store = mockStoreProvider({});
store.dispatch = jest.fn();
let wrapper: ReactWrapper;

describe('Login screen', () => {
  beforeEach(() => {
    wrapper = mount(
      <Provider store={store}>
        <LoginScreen />
      </Provider>,
    );
  });
  afterAll(() => {
    jest.clearAllMocks();
  });
  test('Should match snapshot', () => {
    expect(wrapper).toMatchSnapshot();
  });
  test('Should dispatch start login', () => {
    wrapper
      .find('input[name="lEmail"]')
      .simulate('change', {target: {name: 'lEmail', value: 'test@email.com'}});
    wrapper
      .find('input[name="lPassword"]')
      .simulate('change', {target: {name: 'lPassword', value: 'password'}});
    wrapper.find('.login-form-1 form').simulate('submit');
    expect(startLogin).toHaveBeenCalledWith('test@email.com', 'password');
  });
  test('Should not dispatch start register', () => {
    wrapper.find('input[name="rPassword"]').simulate('change', {
      name: 'rPassword',
      value: '12345678',
    });
    wrapper.find('input[name="rConfirmPassword"]').simulate('change', {
      target: {
        name: 'rConfirmPassword',
        value: '123456789',
      },
    });
    wrapper.find('.login-form-2 form').simulate('submit');
    expect(startRegister).toHaveBeenCalledTimes(0);
    expect(Swal.fire).toHaveBeenCalledWith(
      'Error',
      'Password should match.',
      'error',
    );
  });
});
