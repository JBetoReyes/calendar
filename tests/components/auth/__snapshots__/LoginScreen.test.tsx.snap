// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Login screen Should match snapshot 1`] = `
<Provider
  store={
    Object {
      "clearActions": [Function],
      "dispatch": [MockFunction],
      "getActions": [Function],
      "getState": [Function],
      "replaceReducer": [Function],
      "subscribe": [Function],
    }
  }
>
  <Connect(Component)>
    <Component
      startLogin={[Function]}
      startRegister={[Function]}
    >
      <div
        className="container"
      >
        <div
          className="row"
        >
          <div
            className="login-container"
          >
            <div
              className="col-md-6 login-form-1 h-75"
            >
              <h3>
                Login
              </h3>
              <form
                onSubmit={[Function]}
              >
                <div
                  className="form-group"
                >
                  <input
                    className="form-control"
                    name="lEmail"
                    onChange={[Function]}
                    placeholder="Email"
                    type="text"
                    value=""
                  />
                </div>
                <div
                  className="form-group"
                >
                  <input
                    className="form-control"
                    name="lPassword"
                    onChange={[Function]}
                    placeholder="Password"
                    type="password"
                    value=""
                  />
                </div>
                <div
                  className="form-group"
                >
                  <input
                    className="btnSubmit"
                    type="submit"
                    value="Login"
                  />
                </div>
              </form>
            </div>
            <div
              className="col-md-6 login-form-2 h-75"
            >
              <h3>
                Register
              </h3>
              <form
                onSubmit={[Function]}
              >
                <div
                  className="form-group"
                >
                  <input
                    className="form-control"
                    name="rName"
                    onChange={[Function]}
                    placeholder="Name"
                    type="text"
                    value=""
                  />
                </div>
                <div
                  className="form-group"
                >
                  <input
                    className="form-control"
                    name="rEmail"
                    onChange={[Function]}
                    placeholder="Email"
                    type="email"
                    value=""
                  />
                </div>
                <div
                  className="form-group"
                >
                  <input
                    className="form-control"
                    name="rPassword"
                    onChange={[Function]}
                    placeholder="Password"
                    type="password"
                    value=""
                  />
                </div>
                <div
                  className="form-group"
                >
                  <input
                    className="form-control"
                    name="rConfirmPassword"
                    onChange={[Function]}
                    placeholder="Confirm your password"
                    type="password"
                    value=""
                  />
                </div>
                <div
                  className="form-group"
                >
                  <input
                    className="btnSubmit"
                    type="submit"
                    value="Register"
                  />
                </div>
              </form>
            </div>
          </div>
        </div>
      </div>
    </Component>
  </Connect(Component)>
</Provider>
`;
